### PR TITLE
docs: hide primitive demo on mobile devices

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

This PR will hide primitive demo on mobile devices

#### Description of how you validated changes

Tested locally with Chrome

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/styles/docs/example.scss [hunk 1]
```diff
@@ -32,6 +32,12 @@
   top: var(--amplify-space-small);
 }
 
+@media (max-width: 479px) {
+  h2#demo {
```
Comment (zchenwei) [inline docs/src/styles/docs/example.scss:36]: This is to hide the heading written in mdx

**Review by reesscot — APPROVED** [review]
LGTM

File: docs/src/components/Demo.tsx [hunk 2]
```diff
@@ -35,7 +35,10 @@ export const Demo = ({
   };
 
   return (
-    <View className="docs-component-demo">
+    <View
+      className="docs-component-demo"
+      display={{ base: 'none', small: 'block' }}
```
[the remark this answers is not in the document]
Comment (zchenwei) [reply docs/src/components/Demo.tsx:40]: Oh yea, pretty much the same effect. But I am not against too make it to `medium`

**Review by joebuono — APPROVED** [review]
LGTM

**Review by dbanksdesign — APPROVED** [review]
LGTM! One non-blocking comment

**Comment by zchenwei** [comment]
> LGTM! One non-blocking comment

Good call

**Pull request merged**